complement/pipeline.yml

```diff
@@ -0,0 +1,38 @@
+steps:
+  - command:
+      - "env"
+      - "go build -tags *"
+    label: "\U0001F528 Build / :go: 1.13"
+    plugins:
+      - docker#v3.7.0:
+          image: "golang:1.13"
+          mount-buildkite-agent: false
+    retry:
+      automatic:
+        - exit_status: 128
+          limit: 3
+
+  - command:
+      - "env"
+      - "go build -tags *"
+    label: "\U0001F528 Build / :go: 1.15"
+    plugins:
+      - docker#v3.7.0:
+          image: "golang1.15"
+          mount-buildkite-agent: false
+    retry:
+      automatic:
+        - exit_status: 128
+          limit: 3
+          
+  - command:
+      - "env"
+      # gofmt does not change its exit status depending on whether there are
+      # files in need of formatting. We have to use a slightly hacky way of
+      # checking and then printing the diff instead.
+      - "if [ -n "$(gofmt -l .)" ]; then gofmt -d && exit 1; fi"
```

Comment:
Please don't. `gofmt` shouldn't be used as part of a CI pipeline because it can absolutely change depending on the version you're running. We don't even do this on Dendrite and it hasn't bitten us.

Reply:
Aha, this would probably explain why it doesn't change its exit code then :)

Reply:
Indeed.

> In general, we recommend against building systems that check that source code matches the output of a specific version of gofmt. For example, a continuous integration test that fails if any code already checked into a repository is not “properly formatted” is inherently fragile and not recommended. 

https://golang.org/doc/go1.10#gofmt